### DISCUSSION TITLE
Overview: confusing observation terminology (emission, push, return)

### DISCUSSION THREAD

**Discussion by Chealer** [ReactiveX/rxjs#6572] [Report issues other than bug]
**RxJS version:** _This bug is not in the RxJS product._

**Code to reproduce:** _This does not report a behavioral bug._

**Expected behavior:** _This does not report a behavioral bug._

**Actual behavior:** _This does not report a behavioral bug._

**Additional information:**
[The _Creating Observables_ section of the _Observable_ page of _Overview_](https://rxjs.dev/guide/observable#creating-observables) contains:

> The following example creates an Observable to emit the string 'hi' every second to a subscriber.

That is the first time the overview talks about an Observable _emitting_ something, which can be confusing. Yet, [Observable's specification](https://rxjs.dev/api/index/class/Observable) seems to confirm emitting is a correct and probably even the canonical term to describe that. Prior to that point, the overview seems to use the terms _returning_ and _pushing_ synonymously. If my understanding is correct, current documentation uses all of the following interchangeably:

> emitting
> returning
> pushing
> nexting

Given the topic is already quite complicated, it would help to either be more consistent or explicitly indicate that the above do not refer to different concepts (if that is the case).

**Comment by kwonoj** [comment]
Hi, thanks for the opinions, As pinned https://github.com/ReactiveX/rxjs/issues/5970 , we use gh issues as core team's tracking backlog & code-side bug report only. For further tracking, I'll moving these into discussions page.

**Comment by voliva** [comment]
In my mind these are all synonyms, but it just depends on the context or abstraction level I see more fit using one or the other.

- returning: This I would say it's the least correct, but the way I see it's being used in the docs is always in-between quotes (e.g. `// "return" another value`) when comparing it to functions which return only one value. I think it can be useful for people who's never seen an Observable to understand what scope this new concept covers.
- pushing: This concept is being used when thinking on terms of "pull vs push" - You pull values from a Function or an iterator, and you get values pushed from a promise or an observable. This is just another perspective.
- emitting: This is probably the concept specific for observables.
- nexting: An observable you can call `.next(`, `.error(` or `.complete(`. Nexting is just emiting new values through `.next(`.

I can see how it might be confusing, but I think each one of them can be used depending on the context you're in. Are you trying to get someone that only has used regular functions into Observables? Then explaining it initially in terms of "return" could make sense. Are you explaining it to someone who's played with promises and iterators and is familiar with the pull/push concepts? Then `push` makes all the sense. Are you abstracting over the Observable itself, then `emit` is the right word. Are you down to the implementation level? then `.next()` also works.

**Reply by Chealer** [reply]
"returning" is not always used between quotes (see [the Pull versus Push section](https://rxjs.dev/guide/observable#pull-versus-push) for example).

I don't know about the first 3, but "nexting" should never be used. "next" is not a verb.
